src/backend/metal/CommandBufferMTL.mm

```diff
@@ -304,14 +304,15 @@ void EndSubpass() {
 
                 case Command::DrawElements: {
                     DrawElementsCmd* draw = mCommands.NextCommand<DrawElementsCmd>();
+                    size_t formatSize = IndexFormatSize(lastRenderPipeline->GetIndexFormat());
 
                     ASSERT(encoders.render);
                     [encoders.render
                         drawIndexedPrimitives:lastRenderPipeline->GetMTLPrimitiveTopology()
                                    indexCount:draw->indexCount
                                     indexType:lastRenderPipeline->GetMTLIndexType()
                                   indexBuffer:indexBuffer
-                            indexBufferOffset:indexBufferOffset
+                            indexBufferOffset:indexBufferOffset + draw->firstIndex * formatSize
```

Comment:
This makes the name of the local variable `indexBufferOffset` slightly misleading. Maybe it could be `indexBufferBaseOffset` or something?